### ISSUE TITLE
Index issue for concatenated dataframe

### ISSUE THREAD

**Issue by scottpix23** [vaexio/vaex#605]
I used `vaex.open()` or `vaex.open_many()` to read multiple parquet files into one single concatenated df.  When I do train/test split by applying `df.ml.train_test_split()`, the following error showed up:

```
~/vpy3/lib/python3.6/site-packages/vaex/column.py in trim(self, i1, i2)
    193                 offset += len(self.expressions[i].df)
    194                 i += 1
--> 195                 expressions.append(self.expressions[i])
    196             # if stop <= offset + len(self.expressions[i])
    197             expressions.append(self.expressions[i][start-offset:stop-offset+1])
```
**IndexError: list index out of range**

When I tried to access part of the dataframe by `df[:500]`, the error message said:

```
~/vpy3/lib/python3.6/site-packages/vaex/dataframe.py in add_column(self, name, f_or_array, dtype)
   2929                     if len(self) == len(ar):
   2930                         raise ValueError("Array is of length %s, while the length of the DataFrame is %s due to the filtering, the (unfiltered) length is %s." % (len(ar), len(self), self.length_unfiltered()))
-> 2931                 raise ValueError("array is of length %s, while the length of the DataFrame is %s" % (len(ar), self.length_original()))
   2932             # assert self.length_unfiltered() == len(data), "columns should be of equal length, length should be %d, while it is %d" % ( self.length_unfiltered(), len(data))
   2933             valid_name = vaex.utils.find_valid_name(name)
```
**ValueError: array is of length 501, while the length of the DataFrame is 500**

When I directly train the whole df with IncrementalPredictor, it also failed. 

I guess this issue is absolutely related to issue [563](https://github.com/vaexio/vaex/issues/563) as `df.head()` gave the same error message in issue 563.

Please take a look and fix this issue.
Thanks.

**Comment by maartenbreddels** [comment]
Thansk for the report, this is fixed in #590, but not released yet (should be in vaex v3.0)

**Comment by JovanVeljanoski** [comment]
Closed as new release was made
